chainregistry.go

```diff
@@ -751,3 +760,43 @@ func initNeutrinoBackend(chainDir string) (*neutrino.ChainService, func(), error
 
 	return neutrinoCS, cleanUp, nil
 }
+
+func parseHeaderStateAssertion(state map[string]string) (
+	*headerfs.FilterHeader, error) {
+
+	if len(state) == 0 || state == nil {
+		return nil, nil
+	}
+
+	var headerStateAssertion = &headerfs.FilterHeader{}
+	heightArg, ok := state["height"]
+	if !ok {
+		return nil, fmt.Errorf("unable to create neutrino " +
+			"light client: required parameter height " +
+			"missing from header state assertion")
+	}
+	parsedHeight, err := strconv.ParseUint(heightArg, 10, 32)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create neutrino "+
+			"light client: height parameter from header "+
+			"state assertion is of wrong type: %v", err)
+	}
+	headerStateAssertion.Height = uint32(parsedHeight)
+
+	hashArg, ok := state["hash"]
+	if !ok {
+		return nil, fmt.Errorf("unable to create neutrino " +
+			"light client: required parameter hash " +
+			"missing from header state assertion")
+	}
+	hash, err := chainhash.NewHashFromStr(hashArg)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create neutrino "+
+			"light client: errored parsing hash from "+
+			"header state assertion: %v", err)
+	}
+	headerStateAssertion.FilterHash = *hash
```

Comment:
style nit: initialize and return the struct inline at the end of the method, rather than setting each field independently :)

Reply:
fixed